libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -606,26 +606,28 @@ export class WorkspaceInstallManager extends BaseInstallManager {
       const tempNodeModulesPath: string = `${subspace.getSubspaceTempFolder()}/node_modules`;
       const modulesFilePath: string = `${tempNodeModulesPath}/${RushConstants.pnpmModulesFilename}`;
 
-      const modulesContent: string = FileSystem.readFile(modulesFilePath);
-      const yamlContent: IPnpmModules = yaml.load(modulesContent, { filename: modulesFilePath });
-      const { hoistedDependencies } = yamlContent;
-      const subspaceProject: RushConfigurationProject = subspace.getProjects()[0];
-      const projectNodeModulesPath: string = `${subspaceProject.projectFolder}/node_modules`;
-      for (const value of Object.values(hoistedDependencies)) {
-        for (const [filePath, type] of Object.entries(value)) {
-          if (type === 'public') {
-            // If we don't already have a symlink for this package, create one
-            if (!fs.existsSync(`${projectNodeModulesPath}/${filePath}`)) {
-              // Ensure origin folder exists
-              const parentDirSep: string[] = `${projectNodeModulesPath}/${filePath}`.split(path.sep);
-              parentDirSep.pop();
-              const parentDir: string = parentDirSep.join(path.sep);
-              await FileSystem.ensureFolderAsync(parentDir);
-              BaseLinkManager._createSymlink({
-                linkTargetPath: `${tempNodeModulesPath}/${filePath}`,
-                newLinkPath: `${projectNodeModulesPath}/${filePath}`,
-                symlinkKind: SymlinkKind.Directory
-              });
+      if (await FileSystem.existsAsync(modulesFilePath)) {
+        const modulesContent: string = FileSystem.readFile(modulesFilePath);
```

Comment:
If we're doing `existsAsync()` then we should definitely also use `readFileAsync` here

Reply:
changed